docs/modules/beehive.md

```diff
@@ -0,0 +1,117 @@
+# Beehive
+
+## Beehive Overview
+Beehive is a messaging framework based on go-channels for communication between modules of KubeEdge. A module registered with beehive can communicate with other beehive modules if the name with which other beehive module is registered or the name of the group of the module is known.
+Beehive supports following module operations :
+1) Add Module
+2) Add Module to a group
+3) CleanUp (remove a module from beehive core and all groups)
+
+Beehive support following message operations : 
+1) Send to a module/group
+2) Receive by a module
+3) Send Sync to a module/group
+4) Send Response to a sync message
+
+## Message Format
+Message has 3 parts 
+  1) Header :  
+       1) ID : message ID (string)
+       2) ParentID : if it is a response to a sync message then parentID exists (string)
+       3) TimeStamp : time when message was generated (int)
+       4) Sync : flag to indicate if message is of type sync (bool)
+  2) Route : 
+       1) Source : origin of message (string)
+       2) Group : where message has to be broadcasted (string)
```

Comment:
the group to which the message has to be broadcasted.

Reply:
Done